app/src/main/java/de/westnordost/streetcomplete/data/location/RecentLocationStore.kt

```diff
@@ -0,0 +1,40 @@
+package de.westnordost.streetcomplete.data.location
+
+import android.location.Location
+import de.westnordost.streetcomplete.util.ktx.toLatLon
+import de.westnordost.streetcomplete.util.math.flatDistanceTo
+import kotlin.math.abs
+
+class RecentLocationStore {
+    private val recentLocations: MutableList<Location> = mutableListOf()
```

Comment:
Could you change the type to [`ArrayDeque`](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.collections/-array-deque/)?
```suggestion
    private val recentLocations = ArrayDeque<Location>()
```

It really puzzles me that in `add()`, `recentLocations.removeFirst()` in a loop  is faster than `removeAll` with a filter lambda. Anyway, `ArrayDeque` is optimized for adding and removing elements at the front and at the end.

Reply:
> It really puzzles me that in add(), recentLocations.removeFirst() in a loop is faster than removeAll with a filter lambda

There was a huge but reproducible difference in a test I did, I also hadn't expected this.